API.md

```diff
@@ -1,99 +1,409 @@
 
-# Key algorithms supported by @hapi/jwt
+const badSecretResponse = verifyToken(decodedToken, 'some_unshared_secret');
 
-- public: ['RS256', 'RS384', 'RS512', 'PS256', 'PS384', 'PS512', 'ES256', 'ES384', 'ES512']
-- rsa: ['RS256', 'RS384', 'RS512', 'PS256', 'PS384', 'PS512']
-- hmac: ['HS256', 'HS384', 'HS512']
-- none: ['none]
+// Get response of a unsuccessful verification due to wrong iss
+
+const badIssResonse = verifyToken(decodedToken, 'some_shared_secret', { iss: 'urn:issuer:different_test' });
+
+// Display results to console
+
+console.dir(
+    {
+        token,
+        decodedToken,
+        validResponse,
+        badSecretResponse,
+        badIssResonse
+    },
+    { depth: null }
+);
+```
+Displays the following to the console:
+```js
+{
+  token: 'eyJhbGciOiJIUzUxMiIsInR5cCI6IkpXVCJ9.eyJhdWQiOiJ1cm46YXVka...', // Will vary based on time
+  decodedToken: {
+    token: 'eyJhbGciOiJIUzUxMiIsInR5cCI6IkpXVCJ9.eyJhdWQiOiJ1cm46YXVka...', // Will vary based on time
+    decoded: {
+      header: { alg: 'HS512', typ: 'JWT' },
+      payload: {
+        aud: 'urn:audience:test',
+        iss: 'urn:issuer:test',
+        user: 'some_user_name',
+        group: 'hapi_community',
+        iat: 1600604562, // Will vary based on time
+        exp: 1600618962 // Will vary based on time
+      },
+      signature: 'yh3ASEIrgNJZn...' // Will vary based on time
+    },
+    raw: {
+      header: 'eyJhbGciOiJIUzUxMiIsInR5cCI6IkpXVCJ9',
+      payload: 'eyJhdWQiOiJ1cm46YXVka...', // Will vary based on time
+      signature: 'yh3ASEIrgNJZn...' // Will vary based on time
+    }
+  },
+  validResponse: { isValid: true },
+  badSecretResponse: { isValid: false, error: 'Invalid token signature' },
+  badIssResonse: { isValid: false, error: 'Token payload iss value not allowed' }
+}
+```
+### `generate(payload, secret, [options]`)
+Generates a token as a string where:
+- `payload` - Object that contains [Registered Claim Names](#registered-claim-names) (optional) and additional credentials (optional).  While both [Registered Claim Names](#registered-claim-names) and additional credentials are optional, an empty payload `{}`, would result in a key that only has an `iat` of now. This would make a token that is valid for one second and containing no other information.
+- `secret` - String or buffer that creates signature **or** object where:
+    - `key` - String or buffer that creates signature.
+    - `algorithm`- String containing an accepted [algorithm](#Key-algorithms-supported-by-jwt) to be used.  Default is `'HS256'`.
+- `options` - Optional configuration object with the following:
+    - `header` - Object to put addtional key/value pairs in the header of the token in addition to `alg` and `typ`.
+    - `encoding` - String to set what encoding is used when token is base 64 URL encoded.  Default is `'utf8'`. Encoding done by [@hapi/b64](https://hapi.dev/module/b64/api/?v=5.0.0#base64urlencodevalue).
```

Comment:
`encoding` is not actually supported, see #21.

Reply:
Thank you, it appeared to me that it was just passing the encoding onto b64, was slightly confused why there was not a test for this option, know I know why.

I have removed this line from the API.md

Reply:
It can be added back when #21  is resolved.